homeassistant/components/modem_callerid/button.py

```diff
@@ -0,0 +1,49 @@
+"""Support for Phone Modem button."""
+from __future__ import annotations
+
+from phone_modem import PhoneModem
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_platform
+
+from .const import DATA_KEY_API, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: entity_platform.AddEntitiesCallback,
+) -> None:
+    """Set up the Modem Caller ID sensor."""
+    api = hass.data[DOMAIN][entry.entry_id][DATA_KEY_API]
+    async_add_entities(
+        [
+            PhoneModemButton(
+                api,
+                entry.data[CONF_DEVICE],
+                entry.entry_id,
+            )
+        ]
+    )
+
+
+class PhoneModemButton(ButtonEntity):
+    """Implementation of USB modem caller ID button."""
+
+    def __init__(self, api: PhoneModem, device: str, server_unique_id: str) -> None:
+        """Initialize the button."""
+        self.entity_description = ButtonEntityDescription(
+            key="phone_modem_reject",
+            name="Phone Modem Reject",
+            icon="mdi:phone-hangup",
+        )
```

Comment:
Entity description isn't really used here (which becomes odd, as you now have add keys and such).

We can just use entity class attributes instead.

```suggestion
    """Implementation of USB modem caller ID button."""
    
    _attr_icon = "mdi:phone-hangup"
    _attr_name = "Phone Modem Reject"

    def __init__(self, api: PhoneModem, device: str, server_unique_id: str) -> None:
        """Initialize the button."""
```